design.md

```diff
@@ -0,0 +1,94 @@
+
+
+## Not in scope
+
+* Managing members of the Google Groups. If members need to be added to the Google Group, this will need to happen via the Google Groups UI.
+* Creating/Updating events on groups.io
+* Using Groups from groups.io.
+
+## Format of event files (`y2c.yml`)
+
+```yaml
+id: 'event-id'
+organizer: 'dan@linuxfoundation.com' # email address of the organizer
+calendarId: 'calendar-id'           # Calendar Identifier
+title: 'Event title'
+description: 'Event Description'
+location: 'zoom-url'                # Are we using the Zoom URL as location?
+start:
+  date: '2020-02-10'                # Event start date
+  time: '9:00am'                    # Event start time
+  timeZone: 'America/New_York'
+end:
+  date: '2020-02-10'                # Event end date
+  time: '11:00am'                   # Event end time
+  timeZone: 'America/New_York'
+attendees:
+  - email: joe@example.com          # Guest email
+    name: 'Joe User'                # Guest name (optional)
+  - email: jane@example.com
+    name: 'Jane User'
+reminders:
+  - method: 'email'                 # Send email reminder 10 minutes before event
+    minutes: 10
+  - method: 'popup'                 # Show popup reminder 1 hour before event
+    minutes: 60
+```
+
+#### Notes
+* The user will have to specify an event ID, so changes to the `y2c.yml` file trigger an update on Google Calendar, instead of creating a new event. The user will have to remember not to change the eventId once set, otherwise a new event would be created on Google Calendar.
+* The user will have to specify the Calendar Identifier where the event will be created. The organizer must have access to that calendar.
+* Not sure we care about setting attendee names.
+* Not sure we want to allow users to configure the event's reminders or we want to set the same policy for all events.
+
+## Authorizing requests to the Google Calendar API
+
+The Google Calendar API uses OAuth 2.0, in order to make requests to it in a GitHub action we will have to use [OAuth 2.0 for Server to Server Applications](https://developers.google.com/identity/protocols/OAuth2ServiceAccount). This requires creating a _Service Account_. If we want this _Service Account_ to create events in other users' calendars or on behalf of other users we will have to grant _domain-wide authority_ to the _Service Account_.
+
+At the end of this process a private key file will be generated, the contents of this file will have to be pasted in the [secrets section](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets) on github for each repo so it can be used by the Github Action.
```

Comment:
Please look at whether we can configure secrets once per organization instead, as this would be much easier.

Reply:
It's not possible at the moment. It will be build at some point but they haven't said when it will be released:

https://github.community/t5/GitHub-Actions/Method-ability-to-share-secrets-across-multiple-repositories-in/td-p/30958

https://github.community/t5/GitHub-Actions/Secrets-on-Team-and-Organization-level/td-p/29745